src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMenAge = !century
+    ? people.filter(person => person.sex === 'm')
+    : people.filter(person =>
+      person.sex === 'm'
+      && Math.ceil(person.died / 100) === century);
+
+  return onlyMenAge.reduce((total, man) => (
+    total + (man.died - man.born)
+  ), 0) / onlyMenAge.length;
```

Comment:
you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge